dashboard/app/controllers/api/v1/amazon_future_engineer_controller.rb

```diff
@@ -14,16 +14,19 @@ def submit
     return head :forbidden unless current_user&.teacher?
 
     afe_params = submit_params
+    # Retrieve the school to fill in address (and other) details
+    school = School.find_by(id: afe_params['schoolId'])
```

Comment:
Abstracted this up a level because both submissions now need the school address data